pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -371,5 +424,45 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     for (int i = 0; i < nd; i++)
         nd_shape[i] = np_shape[i];
     return nd_shape;
+}
 
+int get_shape_product(int64_t *shape, int nd, int max_nd)

```

Comment:
can we have a docstring for this as it doesn't do exactly as the name suggests.

Reply:
Is `shape_multiplier` a better name?

Reply:
I wanted to add this in the `array_create` function, but I figured that it is unecessary, since it is only going to be used when copying.
On the other hand if I add it to `array_create`, doing multiple copy operations will benefit from it being precalculated.

Reply:
I disagree, the definition of `multiplier` is "a person or thing that multiplies." So this returns something that can be used to multiply the shape?

Reply:
I don't see this function any more though?

Reply:
> I don't see this function any more though?
it name changed to `arr_element_number`.

Reply:
> I disagree, the definition of `multiplier` is "a person or thing that multiplies." So this returns something that can be used to multiply the shape?

it returns the shape after multiplying some of it elements.

Reply:
So it returns the product of some of the elements of a shape? Which is equivalent to what? The stride in a dimension? `get_dimension_stride`?

Reply:
Honestly I don't mind `get_shape_product`, it's just missing the fact that you don't multiply the shape in all dimensions. `get_product_of_shape_subset`? But that's a bit wordy

Reply:
> So it returns the product of some of the elements of a shape? Which is equivalent to what? The stride in a dimension? `get_dimension_stride`?

Maybe we opt for this and mention the subset part in the doc string?

Reply:
I don't understand how `get_dimension_stride` works, since `strides[x]` represents how many bytes to skip to get to the next element in dimension `x`.
How about `arr_subset_len`.

Reply:
>  since strides[x] represents how many bytes to skip to get to the next element in dimension x

bytes or elements? If it's elements then isn't this equal to what you calculate? If not then I still haven't understood what this function is for.

> How about `arr_subset_len`?

This could work. You'd have to make sure the docstring explains how the subset is defined

Reply:
Nevermind! You are right, I got confused with `Numpy`'s definition of strides (tbh it's not that different from `Pyccel`'s)

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/7a82749e269d45df9f955b988c8fb055831cdf45

Reply:
https://github.com/pyccel/pyccel/pull/1128/commits/ca77d841c2ad2651b358ebfc8533e52b35dda1a0